File: task_1/task_1.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

void line(){
    cout << "-------------------" << endl;
}

void value_print(float x, float y){
    cout << "|" << setw(6) << fixed << setprecision(1) << x << " |"
         << setw(8) << fixed << setprecision(2) << y << " |" << endl; 
}

int main() {
    float x = -6;
    float y = 0;

    cout << "Таблица значений функции:" << endl;
    line();
    cout << "|   x   |    y    |" << endl;
    line();

    for(x; x <= -2; x++)
    {
        y = -x*x - 7*x - 10;
        value_print(x, y);
    }

    for(x; x <= 2; x++)
    {
        y = -sqrt(4 - x*x);
        value_print(x, y);
    }

    for(x; x <= 8; x++)
    {
        y = ( log(x) / log(2) ) - 1;
        value_print(x, y);
    }

    for(x; x <= 10; x++)
    {
        y = -2*x + 18;
        value_print(x, y);
    }
    
    line();

    return 0;
}

File: python_solutions/task_1.py
import math

def line():
    print("-------------------")

def value_print(x, y):
    print(f"| {x:.1f} | {y:.2f} |")

def main():
    print("Таблица значений функции:")
    line()
    print("|   x   |    y    |")
    line()

    x = -6.0
    while x <= -2.0:
        y = -0.992 * (x + 3.5) ** 2 + 2.2
        value_print(x, y)
        x += 1.0

    while x <= 2.0:
        try:
            y = -math.sqrt(4 - x**2)
        except ValueError:
            y = float('nan')
        value_print(x, y)
        x += 1.0

    while x <= 8.0:
        try:
            y = math.log(x, 2) - 1
        except ValueError:
            y = float('nan')
        value_print(x, y)
        x += 1.0

    while x <= 10.0:
        y = -x + 9
        value_print(x, y)
        x += 1.0

    line()

if __name__ == "__main__":
    main()